include/net/socket.h

```diff
@@ -43,6 +43,16 @@ struct zsock_pollfd {
 #define ZSOCK_MSG_PEEK 0x02
 #define ZSOCK_MSG_DONTWAIT 0x40
 
+/* socket option for TCP */
+#define SOL_TCP             6
+
+/*  Socket option for TLS */
+#define SOL_TLS             282
```

Comment:
Linux was used here as a reference:

https://github.com/torvalds/linux/blob/master/include/linux/socket.h

Reply:
Please add comment on that.

And I'd say, it's only worse, because Linux' SOL_TLS has absolutely nothing to do with Zephyr's SOL_TLS. So, if I did it, I'd choose not just different numeric value for it, but also a different symbolic name to avoid confusion.